smallrye-config: Converter<Int> throws IllegalStateException

### Describe the bug

Since Quarkus 2.15.0.Final a converter defined as follows leads to a `IllegalStateException`:

```kotlin
import org.eclipse.microprofile.config.spi.Converter

class MyConverter: Converter<Int> {

    override fun convert(value: String): Int {
       return 1
    }
}
```

This is because `Int` is interpreted as an `int` and not an `java.lang.Integer`. The exception is thrown in
 method `validateConverter` (`io/smallrye/config/smallrye-config-core/2.13.0/smallrye-config-core-2.13.0-sources.jar!/io/smallrye/config/ConfigMappingInterface.java:831`).

```bash
[ERROR] 	[error]: Build step io.quarkus.arc.deployment.ConfigBuildStep#generateConfigClasses threw an exception: java.lang.IllegalArgumentException
[ERROR] 	at io.smallrye.config.ConfigMappingInterface.validateConverter(ConfigMappingInterface.java:831)
[ERROR] 	at io.smallrye.config.ConfigMappingInterface.getPropertyDef(ConfigMappingInterface.java:721)
[ERROR] 	at io.smallrye.config.ConfigMappingInterface.getProperties(ConfigMappingInterface.java:699)
[ERROR] 	at io.smallrye.config.ConfigMappingInterface.createConfigurationInterface(ConfigMappingInterface.java:656)
[ERROR] 	at io.smallrye.config.ConfigMappingInterface.access$000(ConfigMappingInterface.java:28)
[ERROR] 	at io.smallrye.config.ConfigMappingInterface$1.computeValue(ConfigMappingInterface.java:33)
[ERROR] 	at io.smallrye.config.ConfigMappingInterface$1.computeValue(ConfigMappingInterface.java:31)
[ERROR] 	at java.base/java.lang.ClassValue.getFromHashMap(ClassValue.java:228)
[ERROR] 	at java.base/java.lang.ClassValue.getFromBackup(ClassValue.java:210)
[ERROR] 	at java.base/java.lang.ClassValue.get(ClassValue.java:116)
[ERROR] 	at io.smallrye.config.ConfigMappingInterface.getConfigurationInterface(ConfigMappingInterface.java:80)
[ERROR] 	at io.smallrye.config.ConfigMappingInterface.getPropertyDef(ConfigMappingInterface.java:764)
[ERROR] 	at io.smallrye.config.ConfigMappingInterface.getProperties(ConfigMappingInterface.java:699)
[ERROR] 	at io.smallrye.config.ConfigMappingInterface.getProperties(ConfigMappingInterface.java:700)
[ERROR] 	at io.smallrye.config.ConfigMappingInterface.getProperties(ConfigMappingInterface.java:700)
[ERROR] 	at io.smallrye.config.ConfigMappingInterface.createConfigurationInterface(ConfigMappingInterface.java:656)
[ERROR] 	at io.smallrye.config.ConfigMappingInterface.access$000(ConfigMappingInterface.java:28)
[ERROR] 	at io.smallrye.config.ConfigMappingInterface$1.computeValue(ConfigMappingInterface.java:33)
[ERROR] 	at io.smallrye.config.ConfigMappingInterface$1.computeValue(ConfigMappingInterface.java:31)
[ERROR] 	at java.base/java.lang.ClassValue.getFromHashMap(ClassValue.java:228)
[ERROR] 	at java.base/java.lang.ClassValue.getFromBackup(ClassValue.java:210)
[ERROR] 	at java.base/java.lang.ClassValue.get(ClassValue.java:116)
[ERROR] 	at io.smallrye.config.ConfigMappingInterface.getConfigurationInterface(ConfigMappingInterface.java:80)
[ERROR] 	at io.smallrye.config.ConfigMappingLoader.getConfigMappingsMetadata(ConfigMappingLoader.java:28)
[ERROR] 	at io.quarkus.deployment.configuration.ConfigMappingUtils.processConfigClass(ConfigMappingUtils.java:86)
[ERROR] 	at io.quarkus.deployment.configuration.ConfigMappingUtils.generateConfigClasses(ConfigMappingUtils.java:59)
[ERROR] 	at io.quarkus.arc.deployment.ConfigBuildStep.generateConfigClasses(ConfigBuildStep.java:285)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:833)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```

In addition: The exception should have more information about this failure.

### How to Reproduce?

`./mvnw clean install`

[reproducer.zip](https://github.com/quarkusio/quarkus/files/10248164/reproducer.zip)

### Output of `java -version`

Java 17

### Quarkus version or git rev

2.15.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -43,7 +43,7 @@
         <microprofile-lra.version>1.0</microprofile-lra.version>
         <microprofile-openapi.version>2.0.1</microprofile-openapi.version>
         <smallrye-common.version>1.13.2</smallrye-common.version>
-        <smallrye-config.version>2.13.0</smallrye-config.version>
+        <smallrye-config.version>2.13.1</smallrye-config.version>
         <smallrye-health.version>3.3.1</smallrye-health.version>
         <smallrye-metrics.version>3.0.5</smallrye-metrics.version>
         <smallrye-open-api.version>3.1.1</smallrye-open-api.version>

--- a/jakarta/rewrite.yml
+++ b/jakarta/rewrite.yml
@@ -567,7 +567,7 @@ recipeList:
       newValue: '3.0'
   - org.openrewrite.maven.ChangePropertyValue:
       key: smallrye-config.version
-      newValue: '3.1.0'
+      newValue: '3.1.1'
   - org.openrewrite.maven.ChangePropertyValue:
       key: smallrye-fault-tolerance.version
       newValue: '6.1.0'

